index.js

```diff
@@ -8,11 +8,33 @@ var split = require('split-buffer')
 var isBuffer = Buffer.isBuffer
 var concat = Buffer.concat
 
-var box = sodium.crypto_secretbox_easy
-var unbox = sodium.crypto_secretbox_open_easy  
+var buff4096plusMACBYTES = Buffer.alloc(4096 + sodium.crypto_secretbox_MACBYTES)
+var buff18plusMACBYTES = Buffer.alloc(18 + sodium.crypto_secretbox_MACBYTES)
+
+function box (ptxt, nonce, key) {
+  if (ptxt.length === 18) {
+   sodium.crypto_secretbox_easy(buff18plusMACBYTES, ptxt, nonce, key)
+   return buff18
+  }
+  if (ptxt.length === 4096) {
+    sodium.crypto_secretbox_easy(buff4096plusMACBYTES, ptxt, nonce, key)
+    return buff4096
+  }
+
+  var ctxt = new Buffer(ptxt.length + sodium.crypto_secretbox_MACBYTES)
+  sodium.crypto_secretbox_easy(ctxt, ptxt, nonce, key)
+  return ctxt
+}
+
+function unbox (ctxt, nonce, key) {
+  var ptxt = ctxt.slice(16) // TODO
+  sodium.crypto_secretbox_open_easy(ptxt, ctxt, nonce, key)
+  return ptxt
+}
+
 
 function unbox_detached (mac, boxed, nonce, key) {
-  return sodium.crypto_secretbox_open_easy(concat([mac, boxed]), nonce, key)
+  return unbox(concat([mac, boxed]), nonce, key)
```

Comment:
just realized that this concat is also a whole nother copy that isn't necessary... but we should be able to use `crypto_secretbox_open_detached` to pass in the mac separately... oh actually now that I see this, I think it means we can reuse buffers to encryption too!

Reply:
i think we can use _some_ buffers again. since we know that some ptxt will be of a certain length, but there are some situation i.e. the end of the stream, where the ptxt coming in might not be 4096 bytes so you'd need to create a new buffer in those situations. i'll keep fiddling :)

tx for the comment!